### PR TITLE
Videoplayer

### PR DESCRIPTION
**Pull request by OmniBlade**
Base classes for video player subsystem.

### CODE REVIEW

File: src/game/client/videoplayer/videoplayer.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Update() override;
+
+    virtual void Deinit() {}
+    virtual void Lose_Focus() {}
+    virtual void Regain_Focus() {}
+    virtual int Open(AsciiString title) { return 0; }
+    virtual int Load(AsciiString title) { return 0; }
+    virtual VideoStream *First_Stream() { return m_firstStream; }
```
Comment (janisozaur) [inline src/game/client/videoplayer/videoplayer.h:47]: You could make some methods here `const`

File: src/game/client/videoplayer/videoplayer.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+ *            LICENSE
+ */
+#include "videoplayer.h"
+#include "videostream.h"
+#include <cstddef>
+
+#ifndef THYME_STANDALONE
+VideoPlayer *&g_theVideoPlayer = Make_Global<VideoPlayer *>(0x00A2C0AC);
+#else
+VideoPlayer *g_theVideoPlayer = nullptr;
```
Comment (janisozaur) [inline src/game/client/videoplayer/videoplayer.cpp:22]: Is it being initialised anywhere?

Comment (OmniBlade) [reply src/game/client/videoplayer/videoplayer.cpp:22]: Its initialised in GameEngine::Init which is unfinished so the original code currently initialises it.

File: src/game/client/videoplayer/videoplayer.h (hunk 1 above)
Comment (OmniBlade) [reply src/game/client/videoplayer/videoplayer.h:47]: I'll double check the original prototypes, its possible some are intended to be const, First_Stream should almost certainly be const from the look of it. The others will have more complex final implementations in a child class though, the original has an implementation built around the bink library which we'll almost certainly want to replace with the ffmpeg libs.

**Pull request merged**